File: examples/features/authentication/python/authentication_example.py
# authentication_example.py - Example of using authentication in Python

from RobotRaconteur.Client import *

url = 'rr+tcp://localhost:53226?service=authentication_example'

# Create the user credentials
username = "testuser1"
cred = {"password": RR.VarValue("testpass1", "string")}

# Connect to the service
c = RRN.ConnectService(url, username, cred)

# Use the service
c.say("Message to secure service")

File: examples/features/authentication/matlab/authentication_example.m
% authentication_example.m - Example of using authentication from Matlab

url = 'rr+tcp://localhost:53226?service=authentication_example';

% Create the user credentials
username = "testuser1";
cred = containers.Map({'password'}, {RobotRaconteurVarValue('testpass1', 'string')});

% Connect to the service
c = RobotRaconteur.ConnectService(url, username, cred);

% Use the service
c.say('Message to secure service');

RobotRaconteur.DisconnectService(c);
